provider/provider.go

```diff
@@ -104,7 +104,7 @@ func providerConfigure(c context.Context, d *schema.ResourceData) (interface{},
 		backoff.WithMaxRetries(0),
 	)
 
-	client.Concurrency = 10
+	client.Concurrency = 15
```

Comment:
I gave the Http client a little headroom so that we can make a few more requests in parallel than the default terraform parallelism of 10..  The standard limit of 10 will still apply in most cases, as we are doing most requests synchronously.